src/ILLink.RoslynAnalyzer/RequiresAnalyzerBase.cs

```diff
@@ -371,7 +383,7 @@ protected bool TargetHasRequiresAttribute (ISymbol member, [NotNullWhen (returnV
 			}
 
 			// Also check the containing type
-			if (member.IsStatic || member.IsConstructor ()) {
+			if (inheritRucForInstanceMethods || member.IsStatic || member.IsConstructor ()) {
```

Comment:
Rather than use a flag, I think we can keep this method the same, call it, and then check this extra info in the caller